platform-api/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/OAuthAuthenticationService.java

```diff
@@ -236,6 +239,20 @@ public Response invalidate() throws IOException {
                                                                            null)).type(MediaType.TEXT_PLAIN).build();
     }
 
+    @GET
+    @Produces({MediaType.APPLICATION_JSON})
+    @RolesAllowed({"user", "temp_user"})
+    @Path("token")
+    public OAuthToken token(@QueryParam("oauth_provider") String oauth_provider, @Context SecurityContext security)
+            throws Exception {
+        final Principal principal = security.getUserPrincipal();
+        OAuthAuthenticator provider = getAuthenticator(oauth_provider);
```

Comment:
Why <code>principal</code> is defined as <code>final</code> and provider is not?
Also why do you need local variable here?